app/reducers/pages.js

```diff
@@ -68,3 +70,55 @@ export const selectGroupsForHierarchy = createSelector(
     }))
   })
 );
+
+export const selectPageHierarchy = createSelector(
+  (state, props) => props.sections,
+  state => state,
+  (sections, state) =>
+    Object.keys(sections).map(sectionKey =>
+      sections[sectionKey].hierarchySectionSelector(state, {
+        title: sections[sectionKey].title
+      })
+    )
+);
+
+export const selectFlatpageForPages = createSelector(
+  selectPageBySlug,
+  (state, props) => props.pageSlug,
+  (selectedPage, pageSlug) => ({
+    selectedPage,
+    selectedPageInfo: {
+      actionGrant: selectedPage.actionGrant || [],
```

Comment:
shouldn't we always at least return an empty array from the backend?

Reply:
Good point